build_tools/ci/cpu_comparison/run.py

```diff
@@ -758,6 +758,31 @@ def run(self, config):
                 lower_to_aie_pipeline="objectFifo",
             )
 
+        # Test(s) of the form matmul(A,B) + truncf(C) where A:MxK, B:KxN and C:MxN
+        template_name = test_files_dir / "matmul_truncf_32x32x32_bf16_bf16.mlir"
+        identity_mat = np.eye(32, dtype=np.float32)
+        ones = np.ones(32 * 32, dtype=np.float32).reshape([32, 32])
```

Comment:
Because truncf(192 * 2) = 192 * 2, this isn't testing that truncf is doing the right thing (the compiler might just be ignoring it!). Can you choose a matmul where truncf is required, maybe

lhs = ones * 101
rhs = identity_mat * 3 

truncf(101 * 3) = 302  (because 101*3 = 303, and 303 is in range [256, 512), the truncf round it down to the nearest power of 2. 

 nit: can you make the tensor have sizes m=n=k=8 (just to keep it as small and simple as possible)

Reply:
Thanks! I've made the changes. 8x8x8 was failing compilation so added the change required for that too. Please take a look.